scalding-core/src/test/scala/com/twitter/scalding/FileSourceTest.scala

```diff
@@ -216,13 +216,8 @@ class FileSourceTest extends WordSpec with Matchers {
       pathIsGood("test_data/2013/04/") shouldBe false
     }
 
-    "reject an empty directory" in {
-      pathIsGood("test_data/2013/05/") shouldBe false
-      pathIsGood("test_data/2013/05/*") shouldBe false
-    }
-
-    "reject a directory with only _SUCCESS when specified as a glob" in {
-      pathIsGood("test_data/2013/06/*") shouldBe false
+    "accept a directory with only _SUCCESS when specified as a glob" in {
```

Comment:
So, SuccessFileSource had a very complicated pathIsGood contract, and we had some (but not complete) tests to verify that contract, including its oddities that were maybe not desirable but were part of the contract, and we did encode them as tests (which is good).

This PR certainly changes that contract, though I think for the better, and it updates the tests to match.

That said, all this "this is only a best effort / sanity check" approach is a broken window, I think longer term we should move towards an explicit and correct implementation of FilleSource that doesn't rely on globbing oddities. Eg the user says "I want to read these files" and if they aren't there, that's an error. Instead, currently, the users says "I want to read whatever matches this gob, and maybe some verification on the results of that glob" which has the issue of not letting us verify things that don't match the glob at all.